great_expectations/checkpoint/checkpoint.py

```diff
@@ -82,105 +101,216 @@ def __init__(
+            # Next two fields are for LegacyCheckpoint configuration
+            "validation_operator_name": validation_operator_name,
+            "batches": batches,
+        }
+        self._config_kwargs = config_kwargs
+
+    # TODO: Add eval param processing using new TBD parser syntax and updated EvaluationParameterParser and
+    #  parse_evaluation_parameters function (e.g. datetime substitution or specifying relative datetimes like "most
+    #  recent"). Currently, environment variable substitution is the only processing applied to evaluation parameters,
+    #  while run_name_template also undergoes strftime datetime substitution
+    @usage_statistics_enabled_method(
+        event_name="checkpoint.run",
+        args_payload_fn=get_checkpoint_run_usage_statistics,
+    )
+    def run(
+        self,
+        template_name: Optional[str] = None,
+        run_name_template: Optional[str] = None,
+        expectation_suite_name: Optional[str] = None,
+        batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
+        action_list: Optional[List[dict]] = None,
+        evaluation_parameters: Optional[dict] = None,
+        runtime_configuration: Optional[dict] = None,
+        validations: Optional[List[dict]] = None,
+        profilers: Optional[List[dict]] = None,
+        run_id: Optional[Union[str, RunIdentifier]] = None,
+        run_name: Optional[str] = None,
+        run_time: Optional[Union[str, datetime.datetime]] = None,
+        result_format: Optional[Union[str, dict]] = None,
+        expectation_suite_ge_cloud_id: Optional[str] = None,
+    ) -> CheckpointResult:
+        assert not (run_id and run_name) and not (
+            run_id and run_time
+        ), "Please provide either a run_id or run_name and/or run_time."
+
+        run_time = run_time or datetime.datetime.now()
+        runtime_configuration = runtime_configuration or {}
+        result_format = result_format or runtime_configuration.get("result_format")
+
+        batch_request = get_batch_request_as_dict(batch_request=batch_request)
+        validations = get_validations_with_batch_request_as_dict(
+            validations=validations
         )
-        self._config = checkpoint_config
-        self._substituted_config = None
 
-    @property
-    def name(self) -> str:
-        return self._name
+        runtime_kwargs: dict = {
+            "template_name": template_name,
+            "run_name_template": run_name_template,
+            "expectation_suite_name": expectation_suite_name,
+            "batch_request": batch_request or {},
+            "action_list": action_list or [],
+            "evaluation_parameters": evaluation_parameters or {},
+            "runtime_configuration": runtime_configuration or {},
+            "validations": validations or [],
+            "profilers": profilers or [],
+            "expectation_suite_ge_cloud_id": expectation_suite_ge_cloud_id,
+        }
```

Comment:
Same here

Reply:
@cdkini I added the `Attribute` class -- provides the dict/dot notation.  We can improve on it later, if desired.